### ISSUE TITLE
opposite sign of ground heat flux in urban models with Noah-MP land surface model 

### ISSUE THREAD

**Issue by tslin2** [wrf-model/WRF#1921]
https://github.com/NCAR/hrldas/issues/114#issue-1931501060

**Comment by tslin2** [comment]
The ground heat flux in Noah-MP is downwards positive from surface, but it is downwards negative from surface in Noah land surface model (LSM).

To keep the same direction of ground heat flux between urban models (SLUCM, BEP, or BEM) and the land surface model (Noah or Noah-MP) is required to calculate surface energy balance at the grid mean level.

In Noah LSM, the ground heat flux direction is the same as urban models, but differ in Noah-MP.

To make the same direction using SLUCM with Noah-MP LSM at the grid mean level.
[hrldas/urban/wrf/NoahmpUrbanDriverMainMod.F](https://github.com/NCAR/hrldas/blob/f550db9f22c11acbf7aec607904bc22781425e15/urban/wrf/NoahmpUrbanDriverMainMod.F#L530)

Line 530 in [f550db9](https://github.com/NCAR/hrldas/commit/f550db9f22c11acbf7aec607904bc22781425e15)

 GRDFLX(I,J)   = FRC_URB2D(I,J) * (G_URB) + (1-FRC_URB2D(I,J)) * GRDFLX(I,J)        ![W/m/m]   
should revise to
GRDFLX(I,J) = FRC_URB2D(I,J) * (-1.0 * G_URB) + (1-FRC_URB2D(I,J)) * GRDFLX(I,J) ![W/m/m]

and for BEP, BEM
[hrldas/urban/wrf/NoahmpUrbanDriverMainMod.F](https://github.com/NCAR/hrldas/blob/f550db9f22c11acbf7aec607904bc22781425e15/urban/wrf/NoahmpUrbanDriverMainMod.F#L759)

Line 759 in [f550db9](https://github.com/NCAR/hrldas/commit/f550db9f22c11acbf7aec607904bc22781425e15)

 grdflx(i,j)   = (1.-frc_urb2d(i,j))*grdflx_rural(i,j)+ frc_urb2d(i,j)*grdflx_urb(i,j) 
should revise to
grdflx(i,j) = (1.-frc_urb2d(i,j)) * grdflx_rural(i,j)+ frc_urb2d(i,j) * grdflx_urb(i,j) * -1.0

**Comment by weiwangncar** [comment]
@cenlinhe Would you like to comment on this post?

**Comment by cenlinhe** [comment]
Yes, this is a bug. I will look into more detail and try to fix it in the next 1~2 weeks.

**Comment by cenlinhe** [comment]
I confirm that this is a bug: https://github.com/NCAR/hrldas/issues/114#issue-1931501060
I have fixed it in the offline Noah-MP and urban model systems. I am going to submit a PR for WRF system bug fix.
